src/Sarif.WorkItems/SarifWorkItemsExtensions.cs

```diff
@@ -103,20 +104,19 @@ private static string ConstructFullRuleIdentifier(ReportingDescriptor reportingD
             return fullRuleIdentifier;
         }
 
-        public static string CreateWorkItemDescription(this SarifLog log)
+        public static string CreateWorkItemDescription(this SarifLog log, Uri locationUri)
         {
-            Dictionary<string, int> resultCountsByTool = ComputeToolResultCounts(log);
-            StringBuilder templateText = new StringBuilder(@"This bug contains results for the below tool(s) and issue(s).  It was filed automatically.");
-            templateText.AppendLine();
-            templateText.AppendLine();
+            Dictionary<Run, int> resultCountsByRun = ComputeRunResultCounts(log);
+            StringBuilder templateText = new StringBuilder();
 
-            foreach (KeyValuePair<string, int> toolResult in resultCountsByTool)
+            foreach (KeyValuePair<Run, int> run in resultCountsByRun)
```

Comment:
>resultCountsByRun [](start = 51, length = 17)

This helper looks like it creates an undesirable run-on sentence when used with multiple tools. 

This work items contains X Tool1 issues detected in 'Blah'. Click the 'Scans' tab to review results. This work items contains X Tool2 issues detected in 'Baz'. Click the 'Scans' tab to review results. This work items contains X Tool3 issues detected in 'Bar'. Click the 'Scans' tab to review results. 

Instead, consider aggregating the result count and gathering all the tool names into a single sentence. 

This work item contains X 'FxCop', 'BinSkim' and 'Fortify' issue(s) detected in 'blah.cs' (and other locations). Click the 'Scans' tab to review results."

Write a little helper that will concatenate your tool names. Be sure to enquote these with apostrophes. Be sure to enquote the location with apostrophes as well. The text 'and other locations' would only be appended in the multi-run case.